src/asm/lexer.c

```diff
@@ -1398,11 +1391,60 @@ static char const *readInterpolation(void)
 	return NULL;
 }
 
+static int appendMacroArg(char const *str, int i)
+{
+	while (*str && i < sizeof(yylval.tzString)) {
+		int c = *str++;
+
+		if (c != '\\') {
+			yylval.tzString[i++] = c;
+			continue;
+		}
+
+		c = *str++;
+
+		switch (c) {
+		case '\\': /* Return that character unchanged */
+		case '"':
+		case '{':
+		case '}':
+			break;
+		case 'n':
+			c = '\n';
+			break;
+		case 'r':
+			c = '\r';
+			break;
+		case 't':
+			c = '\t';
+			break;
+
+		/*
+		 * Line continuations and macro args were already
+		 * handled while reading the macro args
+		 */
+
+		case '\0': /* Can't really print that one */
+			error("Illegal character escape at end of macro arg\n");
+			yylval.tzString[i++] = '\\';
+			return i;
+		default:
+			error("Illegal character escape '%s'\n", print(c));
```

Comment:
It may be good to add a comment here that macro args should not be reached here, in case it happens later due to a bug, and someone wonders why macro args are reported as being invalid.